src/gpuarray_buffer_cuda.c

```diff
@@ -1210,16 +1306,16 @@ static gpukernel *cuda_newkernel(gpucontext *c, unsigned int count,
     ctx->refcnt++;
     cuda_exit(ctx);
     TAG_KER(res);
-    psb = memdup(&sb, sizeof(strb));
-    if (psb == NULL) {
-      cuda_freekernel(res);
-      strb_clear(&sb);
-      FAIL(NULL, GA_MEMORY_ERROR);
+    psrc = memdup(&src, sizeof(strb));
+    if (psrc != NULL) {
+      /* One of the refs is for the cache */
+      res->refcnt++;
```

Comment:
Because I would have to teach the cache how to add references (which is not possible for just any object) where as now I just have to teach it how to free objects.